Integration/Integration.xcodeproj/project.pbxproj

```diff
@@ -0,0 +1,1379 @@
+				SDKROOT = iphoneos;
+				VERSIONING_SYSTEM = "apple-generic";
+				VERSION_INFO_PREFIX = "";
+			};
+			name = Debug;
+		};
+		0A87D3652175E4A300BD0C4C /* Release */ = {
+			isa = XCBuildConfiguration;
+			baseConfigurationReference = 0A87D4DC2177345B00BD0C4C /* Module-Release.xcconfig */;
+			buildSettings = {
+				ALWAYS_SEARCH_USER_PATHS = NO;
+				CLANG_ANALYZER_NONNULL = YES;
+				CLANG_ANALYZER_NUMBER_OBJECT_CONVERSION = YES_AGGRESSIVE;
+				CLANG_CXX_LANGUAGE_STANDARD = "gnu++14";
+				CLANG_CXX_LIBRARY = "libc++";
+				CLANG_ENABLE_MODULES = YES;
+				CLANG_ENABLE_OBJC_ARC = YES;
+				CLANG_ENABLE_OBJC_WEAK = YES;
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
+				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
+				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
+				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
+				CLANG_WARN_DOCUMENTATION_COMMENTS = YES;
+				CLANG_WARN_EMPTY_BODY = YES;
+				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
+				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
+				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
+				CLANG_WARN_SUSPICIOUS_MOVE = YES;
+				CLANG_WARN_UNGUARDED_AVAILABILITY = YES_AGGRESSIVE;
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				CODE_SIGN_IDENTITY = "iPhone Developer";
+				COPY_PHASE_STRIP = NO;
+				CURRENT_PROJECT_VERSION = 1;
+				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				ENABLE_NS_ASSERTIONS = NO;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_C_LANGUAGE_STANDARD = gnu11;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES_ERROR;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
+				IPHONEOS_DEPLOYMENT_TARGET = 12.0;
+				MTL_ENABLE_DEBUG_INFO = NO;
+				MTL_FAST_MATH = YES;
+				SDKROOT = iphoneos;
+				VALIDATE_PRODUCT = YES;
+				VERSIONING_SYSTEM = "apple-generic";
+				VERSION_INFO_PREFIX = "";
+			};
+			name = Release;
+		};
+		0A87D3D72175E56400BD0C4C /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CODE_SIGN_STYLE = Manual;
+				DEVELOPMENT_TEAM = "";
+				INFOPLIST_FILE = iOSStaticLibraryIntegration/Info.plist;
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+				);
+				PRODUCT_BUNDLE_IDENTIFIER = org.example.iOSStaticLibraryIntegration;
+				PRODUCT_NAME = "$(TARGET_NAME)";
+				PROVISIONING_PROFILE_SPECIFIER = "";
+				SUPPORTED_PLATFORMS = "iphonesimulator iphoneos";
+				TARGETED_DEVICE_FAMILY = "1,2";
```

Comment:
I think these are target settings. I also think we can clean some up: `CODE_SIGN_STYLE` `DEVELOPMENT_TEAM` `LD_RUNPATH_SEARCH_PATHS` `PRODUCT_BUNDLE_IDENTIFIER` `PRODUCT_NAME` `PROVISIONING_PROFILE_SPECIFIER`.

Reply:
@bpoplauschi Product bundle identifier doesn't work for watch os targets.

For that I would like to add `PRODUCT_BUNDLE_IDENTIFIER_ORGANIZATION_PREFIX` or `PRODUCT_BUNDLE_IDENTIFIER_PREFIX` with value `com.deusty.` ( not sure about trailing dot )

Reply:
@lolgear Just out of interest: Why doesn't it work for watchOS targets? (it's been some time since I worked with watchOS... 😬)

But otherwise I'd vote for `PRODUCT_BUNDLE_IDENTIFIER_PREFIX = com.deusty` (without trailing dot).

Reply:
@ffried 
Ok, without trailing dot.

```
xcodebuild clean build -project Integration.xcodeproj -scheme 'watchOSSwiftIntegration' -configuration Release -sdk watchsimulator -destination 'platform=iOS Simulator,name=iPhone XS Max,OS=12.0' | xcpretty -c
```

This action will build framework for iPhone simulator, not for Watch OS Simulator :(

Output:
```
ld: URGENT: building for watchOS Simulator simulator, but linking against dylib (~/Library/Developer/Xcode/DerivedData/Integration-cpyxjtzsyeaofnaanurdqqcwsezf/Build/Products/Release-iphonesimulator/CocoaLumberjackSwift.framework/CocoaLumberjackSwift) built for iOS Simulator. Note: This will be an error in the future.
```

If you look at `Products/` directory in `DerivedData/`, you will notice that CocoaLumberjack is stored in `Releas-iphoneos/`.

Reply:
agree with `PRODUCT_BUNDLE_IDENTIFIER_PREFIX = com.deusty`

about the wrong framework platform being built, I also want to take a look. I had the same issue and decided to add the `Podfile` for the integration.

Reply:
> `ld: URGENT: building for watchOS Simulator simulator, but linking against dylib (~/Library/Developer/Xcode/DerivedData/Integration-cpyxjtzsyeaofnaanurdqqcwsezf/Build/Products/Release-iphonesimulator/CocoaLumberjackSwift.framework/CocoaLumberjackSwift) built for iOS Simulator. Note: This will be an error in the future.`
That warning has been there since at least watchOS 2... I wonder whether we can just ignore it (until it actually becomes an error, at which point I hope there's a fix for the `xcodebuild` command). AFAIK it does not happen when building against the device (or doing an archive for that matter).

> `-destination 'platform=iOS Simulator,name=iPhone XS Max,OS=12.0'`
Just wondering: Is the "iPhone XS Max" linked to a watch here?

Reply:
Good point!  
xcodebuild command for watchOS simulator app (**NOT** watchOS framework) needs the destination of iOS Simulator with details info (including name (e.g.iPhone XS) ). 
Though, I have been thinking this specification is a strange and special for a long time, but it's true😝(On the other hand, watchOS framework simulator needs watchOS simulator☺️)

Then you can add the following xcodebuild command to .travis.yml for watchOS app simulator.
```
- xcodebuild clean build -project Integration.xcodeproj -scheme 'watchOSSwiftIntegration' -configuration Release -sdk watchsimulator -destination 'platform=iOS Simulator,name=iPhone XS Max,OS=12.0' | xcpretty -c
```

In addition, I confirmed `pod lib lint` passed.

Reply:
```
⚠️  ld: URGENT: building for watchOS Simulator simulator, but linking against dylib (/Users/dmitry/Library/Developer/Xcode/DerivedData/Integration-cpyxjtzsyeaofnaanurdqqcwsezf/Build/Products/Release-iphonesimulator/CocoaLumberjackSwift.framework/CocoaLumberjackSwift) built for iOS Simulator. Note: This will be an error in the future.
```

Reply:
I'd suggest to try an archive of a watchOS app. If it fails, we actually do have an issue. Otherwise I'd really just suggest to ignore the warning for now. The `xcodebuild` command looks fine to me and I suspect the bug is really just in Xcode somewhere.